src/test/util/setup_common.cpp

```diff
@@ -217,7 +217,7 @@ CBlock TestChain100Setup::CreateAndProcessBlock(const std::vector<CMutableTransa
     for (const CMutableTransaction& tx : txns) {
         block.vtx.push_back(MakeTransactionRef(tx));
     }
-    RegenerateCommitments(block);
+    RegenerateCommitments(block, WITH_LOCK(::cs_main, return std::ref(g_chainman.m_blockman)));
```

Comment:
Can we pass directly the result of `LookupBlockIndex(block.hashPrevBlock)` ? Block is already known here. Because AFAIU, the lock isn't held once `RegenerateCommitment` argument are loaded on the stack thus do you have guarantee you're calling the new `blockman.LookupBlockIndex()` with a lock ?

I think so otherwise `LookupBlockIndex`'s lock annotation would yelled but I can't convince myself of it...

Reply:
Right, this is a shitty situation where the lock in this diff is only needed because:
https://github.com/bitcoin/bitcoin/blob/831675c8dccfa6525ffe751da3cc60709c380953/src/validation.h#L944
and
https://github.com/bitcoin/bitcoin/blob/831675c8dccfa6525ffe751da3cc60709c380953/src/validation.h#L840

Then the lock is released as `RegenerateCommitments`'s assembly function prologue is run, and the same lock is taken again when `LookupBlockIndex` is called inside `RegenerateCommitments`.

The original reason why I wanted to just pass in the `blockman` is because I didn't want to impose the constraint of "arg 2 must be the block index of arg 1" on `RegenerateCommitments`'s arguments, which seems like something that might be easy to cause a bug. Perhaps adequate documentation can prevent this, or asserts that only run at test/debug time?